Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -49,14 +49,43 @@ namespace AzToolsFramework
             m_transform = nullptr;
         }
 
-        void EditorComponentBase::SetEntity(AZ::Entity* entity)
+        void EditorComponentBase::PrepareComponentForEntityAddition(AZ::Entity* entity)
         {
-            Component::SetEntity(entity);
-
             if (entity && m_alias.empty())
             {
-                AZ_Assert(GetId() != AZ::InvalidComponentId, "ComponentId is invalid.");
-                m_alias = AZStd::string::format("Component_[%llu]", GetId());
+                AZStd::unordered_set<AZStd::string> serializedIdentifiersMatchingType;
+
+                for (const AZ::Component* componentInEntity : entity->GetComponents())
+                {
+                    if (componentInEntity == this)
+                    {
+                        return;
+                    }
```

Comment:
Should it be 'continue'?

Reply:
This is stopping the function execution if the component is already added to the entity. We already checked that m_alias shouldn't be empty on line 54. So the idea is that if a component already has an alias and its already added to the entity, there's nothing to do here; return. 

I'll add a comment here to make this clearer.